### PR TITLE
#782: Add support for Sonarqube 10.1

### LINKED ISSUE #782: Plugin incompatible with 10.1.0 SonarQube bug
**Describe the bug**
UI does not show up on upgrade to SonarQube 10.1.0

**To Reproduce**
Perform an upgrade to 10.1.0

**Expected behavior**

**Screenshots**

**Software Versions**
 - SonarQube Version: 10.1.0 (build 73491)
 - Plugin Version: 1.15.0-SNAPSHOT from [Pull-Request](https://github.com/mc1arke/sonarqube-community-branch-plugin/pull/754)

**Additional context**
Here is the error message:

```
2023.06.21 11:55:45 WARN  web[][o.s.c.a.AnnotationConfigApplicationContext] Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-org.sonar.server.ws.WebServiceEngine': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.PullRequestWs': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction]: Constructor threw exception; nested exception is java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
2023.06.21 11:55:45 ERROR web[][o.s.s.p.Platform] Background initialization failed. Stopping SonarQube
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-org.sonar.server.ws.WebServiceEngine': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.PullRequestWs': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction]: Constructor threw exception; nested exception is java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:229)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1372)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1222)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:920)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
        at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:187)
        at org.sonar.server.platform.platformlevel.PlatformLevel.start(PlatformLevel.java:80)
        at org.sonar.server.platform.platformlevel.PlatformLevel4.start(PlatformLevel4.java:662)
        at org.sonar.server.platform.PlatformImpl.start(PlatformImpl.java:214)
        at org.sonar.server.platform.PlatformImpl.startLevel34Containers(PlatformImpl.java:194)
        at org.sonar.server.platform.PlatformImpl$AutoStarterRunnable.runIfNotAborted(PlatformImpl.java:362)
        at org.sonar.server.platform.PlatformImpl$1.doRun(PlatformImpl.java:113)
        at org.sonar.server.platform.PlatformImpl$AutoStarterRunnable.run(PlatformImpl.java:346)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.PullRequestWs': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction]: Constructor threw exception; nested exception is java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:229)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1372)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1222)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.addCandidateEntry(DefaultListableBeanFactory.java:1609)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1573)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveMultipleBeans(DefaultListableBeanFactory.java:1439)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1349)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311)
        at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
        ... 21 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7-com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction]: Constructor threw exception; nested exception is java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:315)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:296)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1372)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1222)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.addCandidateEntry(DefaultListableBeanFactory.java:1609)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1573)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveMultipleBeans(DefaultListableBeanFactory.java:1439)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1349)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311)
        at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
        ... 38 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction]: Constructor threw exception; nested exception is java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:224)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:117)
        at org.springframework.beans.factory.support.ConstructorResolver.lambda$instantiate$0(ConstructorResolver.java:307)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:306)
        ... 55 common frames omitted
Caused by: java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/sonar/server/ws/WsUtils. Method 'void writeProtobuf(com.google.protobuf.Message, org.sonar.api.server.ws.Request, org.sonar.api.server.ws.Response)' at index 432 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction.<init>(ListAction.java:63)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211)
        ... 59 common frames omitted
2023.06.21 11:55:45 INFO  web[][o.s.p.ProcessEntryPoint] Hard stopping process
2023.06.21 11:55:46 INFO  web[][c.z.h.HikariDataSource] HikariPool-1 - Shutdown initiated...
2023.06.21 11:55:46 INFO  web[][c.z.h.HikariDataSource] HikariPool-1 - Shutdown completed.
2023.06.21 11:55:46 WARN  web[][o.a.c.l.WebappClassLoaderBase] The web application [ROOT] appears to have started a thread named [Thread-1] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:\n java.base@17.0.7/sun.nio.ch.SocketDispatcher.read0(Native Method)\n java.base@17.0.7/sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:47)\n java.base@17.0.7/sun.nio.ch.NioSocketImpl.tryRead(NioSocketImpl.java:261)\n java.base@17.0.7/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:312)\n java.base@17.0.7/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)\n java.base@17.0.7/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)\n java.base@17.0.7/java.net.Socket$SocketInputStream.read(Socket.java:966)\n java.base@17.0.7/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:484)\n java.base@17.0.7/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:478)\n java.base@17.0.7/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)\n java.base@17.0.7/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1465)\n java.base@17.0.7/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1069)\n java.base@17.0.7/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)\n java.base@17.0.7/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)\n java.base@17.0.7/java.io.BufferedInputStream.read(BufferedInputStream.java:343)\n java.naming@17.0.7/com.sun.jndi.ldap.Connection.run(Connection.java:856)\n java.base@17.0.7/java.lang.Thread.run(Thread.java:833)
```

### PR DESCRIPTION
**Pull request by bluebu**
based on the branch: feature/sonarqube-10_0-support https://github.com/mc1arke/sonarqube-community-branch-plugin/pull/754 https://github.com/mc1arke/sonarqube-community-branch-plugin/issues/753

    1. upgrade sonarqubeVersion from '10.0.0.68432' to '10.1.0.73491'
    2. fixed dbClient.componentDao().insert(DbSession session, ComponentDto item, boolean isMainBranch)
    3. fixed UT code

fixed #782 

Get the Artifacts, see as:

https://github.com/bluebu/sonarqube-community-branch-plugin/actions/runs/5621946627

### CODE REVIEW

File: src/main/java/com/github/mc1arke/sonarqube/plugin/server/CommunityBranchSupportDelegate.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public ComponentDto createBranchComponent(DbSession dbSession, BranchSupport.Com
             .setUuidPath(ComponentDto.UUID_PATH_OF_ROOT)
             .setMainBranchProjectUuid(mainComponentDto.uuid())
             .setCreatedAt(new Date(clock.millis()));
-        dbClient.componentDao().insert(dbSession, componentDto);
+        dbClient.componentDao().insert(dbSession, componentDto, true);
```
Comment (bluebu) [inline src/main/java/com/github/mc1arke/sonarqube/plugin/server/CommunityBranchSupportDelegate.java:101]: I can't make sure this line is right or not

**Comment by qathy** [comment]
Hello,
I've tried to build from the sources, injecting this PR.
Unfortunately, it doesn't compile as is because loadProjectConfiguration has a new signature:
loadProjectConfiguration(DbSession dbSession, String projectUuid);

=> need to change from:
.setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto), branchName))
=> to:
.setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto.uuid()), branchName))

Still need to try if it works fine. This should be done next week.

**Comment by Operativekiwi** [comment]
> Hello, I've tried to build from the sources, injecting this PR. Unfortunately, it doesn't compile as is because loadProjectConfiguration has a new signature: loadProjectConfiguration(DbSession dbSession, String projectUuid);
> 
> => need to change from: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto), branchName)) => to: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto.uuid()), branchName))
> 
> Thanks to that, branch management is back in my SonarQube 10.1 instance.
> 
> Still need to check if MR are working as expected.

I'm also facing an issue with building this, are you able to send the built JAR file?

**Comment by bluebu** [comment]
> 

https://github.com/mc1arke/sonarqube-community-branch-plugin/suites/14469323013/artifacts/842449126

**Comment by bluebu** [comment]
> > Hello, I've tried to build from the sources, injecting this PR. Unfortunately, it doesn't compile as is because loadProjectConfiguration has a new signature: loadProjectConfiguration(DbSession dbSession, String projectUuid);
> > => need to change from: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto), branchName)) => to: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto.uuid()), branchName))
> > Thanks to that, branch management is back in my SonarQube 10.1 instance.
> > Still need to check if MR are working as expected.
> 
> I'm also facing an issue with building this, are you able to send the built JAR file?

Get the Artifacts, see as:

https://github.com/bluebu/sonarqube-community-branch-plugin/actions/runs/5621946627

**Comment by qathy** [comment]
> > Hello, I've tried to build from the sources, injecting this PR. Unfortunately, it doesn't compile as is because loadProjectConfiguration has a new signature: loadProjectConfiguration(DbSession dbSession, String projectUuid);
> > => need to change from: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto), branchName)) => to: .setExcludeFromPurge(isBranchExcludedFromPurge(projectConfigurationLoader.loadProjectConfiguration(dbSession, mainComponentDto.uuid()), branchName))
> > Thanks to that, branch management is back in my SonarQube 10.1 instance.
> > Still need to check if MR are working as expected.
> 
> I'm also facing an issue with building this, are you able to send the built JAR file?

Sorry for the **very late** answer.
I won't deliver a jar file, as it is contained in artifacts officially delivered here: https://github.com/mc1arke/sonarqube-community-branch-plugin/suites/14469323013/artifacts/842449126
But just in case it may help, here is the git diff revealing my changes:
[git_diff_sq_branch.txt](https://github.com/mc1arke/sonarqube-community-branch-plugin/files/12359507/git_diff_sq_branch.txt)

**Comment by keinar** [comment]
Hi all, can anyone here explain why these changes do not merge yet? 
Or someone knows when the support for Sonarqube 10.1 will add to the plugin?

**Comment by nixel2007** [comment]
> Hi all, can anyone here explain why these changes do not merge yet?

Probably because repo owner has real life.

**Comment by jeremytbrun** [comment]
Like many others I've been waiting for this to be merged to support SonarQube 10.1. My question is if I were to pull down the artifact from the build of this PR and use it what potential issues would that introduce down the road once an official new version is released?

https://github.com/mc1arke/sonarqube-community-branch-plugin/actions/runs/5621946976

**Comment by marcelstoer** [comment]
> > Hi all, can anyone here explain why these changes do not merge yet?
> 
> Probably because repo owner has real life.

Yes and no. It's the start of a pattern that often eventually leads to the death of once thriving open-source projects. Been there, done that (on both sides). Once PRs start piling up and go unrecognized by the maintainer(s), the community becomes discouraged and starts loosing interest. The root cause often is the dependency on a single owner/maintainer. If they didn't not recruit other maintainers in time for whatever reason (lack of community interest, own ego, etc.) projects then either die or forks start popping up - and the cycle starts anew.

**Comment by hballangan-mdsol** [comment]
> Yes and no. It's the start of a pattern that often eventually leads to the death of once thriving open-source projects. Been there, done that (on both sides). Once PRs start piling up and go unrecognized by the maintainer(s), the community becomes discouraged and starts loosing interest. The root cause often is the dependency on a single owner/maintainer. If they didn't not recruit other maintainers in time for whatever reason (lack of community interest, own ego, etc.) projects then either die or forks start popping up - and the cycle starts anew.

For those generally interested in this topic I recommend ["Roads and Bridges: The Unseen Labor Behind Our Digital Infrastructure"](https://www.fordfoundation.org/work/learning/research-reports/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure/).

Does anyone at least has a working fork for Sonarqube 10?

**Comment by mc1arke** [comment]
I'll re-create this against master branch once the 10.0 release has been created

**Comment by thoikono** [comment]
Hello,

Thanks for the info above
I am using helm chart 10.2 so SonarQube version is 10.2.
Just to understand
Currently we have https://github.com/mc1arke/sonarqube-community-branch-plugin/releases/download/1.14.0/sonarqube-community-branch-plugin-1.14.0.jar which is not working with SonarQube 10.x.
Correct?
Then we have this one : https://github.com/mc1arke/sonarqube-community-branch-plugin/suites/14469323013/artifacts/842449126 which is supposed to work with Sonarqube 10.x ?
However when I try to curl the artifacts url I get 404 ... 
Any help?

**Comment by randymelo** [comment]
hello people, any try this artifact whit the version 10.2.1?

**Comment by dedece35** [comment]
> https://github.com/mc1arke/sonarqube-community-branch-plugin/suites/14469323013/artifacts/842449126

+1

**Comment by dedece35** [comment]
> hello people, any try this artifact whit the version 10.2.1?

+1

**Pull request closed without merging**